### ISSUE TITLE
4.04.0 does not allow -c and -o on C files

### ISSUE THREAD

**Issue by chambart** [OCamlPro/operf-micro#22]
On 4.04 the build command fails with:

```
operf-micro build
building operf base files

Command return code 2:
  '/home/louis/.opam/4.04.1+dev+bench/bin/ocamlc.opt' -c -I '/home/chambart/hand_test/micro/4.04/.operf/micro/build' -o '/home/chambart/hand_test/micro/4.04/.operf/micro/build/cycles.o' '/home/chambart/hand_test/micro/4.04/.operf/micro/build/cycles.c'
Options -c and -o are incompatible when compiling C files
```

**Comment by lefessan** [comment]
Actually, I am surprised by this error: the code I wrote in `driver/compenv.ml` is supposed to make an exception when the `-o` specifies exactly the object file that will be generated by `ocamlc` (`-o` does not work on C files only because of MSVC if I remember).

**Comment by lefessan** [comment]
Could you print `ocamlc.opt -config` ?

**Comment by chambart** [comment]
What are you suspecting ?

```
$ /home/louis/.opam/4.04.1+dev+bench/bin/ocamlc -config
version: 4.04.1+dev0-2016-11-04
standard_library_default: /home/louis/.opam/4.04.1+dev+bench/lib/ocaml
standard_library: /home/louis/.opam/4.04.1+dev+bench/lib/ocaml
standard_runtime: /home/louis/.opam/4.04.1+dev+bench/bin/ocamlrun
ccomp_type: cc
bytecomp_c_compiler: gcc -std=gnu99 -O2 -fno-strict-aliasing -fwrapv -Wall -D_FILE_OFFSET_BITS=64 -D_REENTRANT -fPIC
bytecomp_c_libraries: -lm  -ldl -lcurses -lpthread                  
native_c_compiler: gcc -std=gnu99 -O2 -fno-strict-aliasing -fwrapv -Wall -D_FILE_OFFSET_BITS=64 -D_REENTRANT
native_c_libraries: -lm  -ldl
native_pack_linker: ld -r  -o 
ranlib: ranlib
cc_profile: -pg
architecture: amd64
model: default
system: linux
asm: as
asm_cfi_supported: true
with_frame_pointers: false
ext_obj: .o
ext_asm: .s
ext_lib: .a
ext_dll: .so
os_type: Unix
default_executable_name: a.out
systhread_supported: true
host: x86_64-unknown-linux-gnu
target: x86_64-unknown-linux-gnu
flambda: false
spacetime: false
safe_string: false
exec_magic_number: Caml1999X011
cmi_magic_number: Caml1999I021
cmo_magic_number: Caml1999O011
cma_magic_number: Caml1999A012
cmx_magic_number: Caml1999Y015
cmxa_magic_number: Caml1999Z014
ast_impl_magic_number: Caml1999M020
ast_intf_magic_number: Caml1999N018
cmxs_magic_number: Caml2007D002
cmt_magic_number: Caml2012T008
```

**Comment by lefessan** [comment]
I was suspecting the computation of `c_object_of_filename`, but it is indeed a bug of the `Makefile`: the  only correct argument of `-o` would be `cycles.o` and not `/home/chambart/hand_test/micro/4.04/.operf/micro/build/cycles.o`, i.e. `ocamlc` generates the file directly in the current directory, so the computation of `c_object_of_filename` is correct in `ocamlc`.

**Comment by chambart** [comment]
Well, it is not a makefile, it is a call from operf-micro. Which means that it needs to `cd` to the right directory before building.

**Comment by nojb** [comment]
Are there plans to fix this ? I would like to use this to perform some benchmarking ...

**Comment by chambart** [comment]
Fixed in 1.1.1
